tests/PHPStan/Rules/Classes/MixinRuleTest.php

```diff
@@ -84,6 +84,15 @@ public function testRule(): void
 				'PHPDoc tag @mixin contains non-object type int.',
 				92,
 			],
+			[
+				'Type projection contravariant MixinRule\Foo in generic type MixinRule\Adipiscing<contravariant MixinRule\Foo> in PHPDoc tag @mixin is conflicting with variance of template type T of class MixinRule\Adipiscing.',
```

Comment:
Two things:

1) Should we call this "type projection"? What exactly is the projection? Is `MixinRule\Adipiscing<Foo>` (the common syntax we use today) also some kind of projection? I like "call-site variance" much better as opposed to "declaration variance".
2) The second part of the message "conflicting with variance of template type T" could specify the exact variance, so it could say:

* `"conflicting with covariance of template type T"`
* `"conflicting with invariance of template type T"`
* `"conflicting with contravariance of template type T"`

Reply:
Good point, I've reworded the messages. Different sources use the term "type projection" rather vaguely, so it's for the best to use "call-site variance", hopefully it should clear any confusion about what the error refers to.